src/Grids/convenience_constructors.jl

```diff
@@ -0,0 +1,680 @@
+    x1boundary = (:east, :west)
+    x2boundary = (:south, :north)
+    domain = Domains.RectangleDomain(
+        Domains.IntervalDomain(
+            Geometry.XPoint{FT}(x_min),
+            Geometry.XPoint{FT}(x_max);
+            periodic = periodic_x,
+            boundary_names = x1boundary,
+        ),
+        Domains.IntervalDomain(
+            Geometry.YPoint{FT}(y_min),
+            Geometry.YPoint{FT}(y_max);
+            periodic = periodic_y,
+            boundary_names = x2boundary,
+        ),
+    )
+    return Meshes.RectilinearMesh(domain, x_elem, y_elem)
+end
+
+#####
+##### Grids
+#####
+
+"""
+    ExtrudedCubedSphereGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        z_elem::Integer,
+        z_min::Real,
+        z_max::Real,
+        radius::Real,
+        h_elem::Integer,
+        n_quad_points::Integer,
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        stretch::Meshes.StretchingRule = Meshes.Uniform(),
+        hypsography::HypsographyAdaption = Flat(),
+        global_geometry::Geometry.AbstractGlobalGeometry = Geometry.ShallowSphericalGlobalGeometry(radius),
+        quad::Quadratures.QuadratureStyle = Quadratures.GLL{n_quad_points}(),
+        h_mesh = Meshes.EquiangularCubedSphere(Domains.SphereDomain{FT}(radius), h_elem),
+        h_topology::Topologies.AbstractDistributedTopology = Topologies.Topology2D(context, h_mesh),
+        horizontal_layout_type = DataLayouts.IJFH,
+        z_mesh::Meshes.IntervalMesh = DefaultZMesh(FT; z_min, z_max, z_elem, stretch),
+        enable_bubble::Bool = false
+    )
+
+A convenience constructor, which builds an
+`ExtrudedFiniteDifferenceGrid`, given:
+
+ - `FT` the float type (defaults to `Float64`) [`Float32`, `Float64`]
+ - `z_elem` the number of z-points
+ - `z_min` the domain minimum along the z-direction.
+ - `z_max` the domain maximum along the z-direction.
+ - `radius` the radius of the cubed sphere
+ - `h_elem` the number of elements per side of every panel (6 panels in total)
+ - `n_quad_points` the number of quadrature points
+ - `device` the [`ClimaComms.device`](@ref)
+ - `context` the [`ClimaComms.context`](@ref)
+ - `stretch` the mesh [`Meshes.StretchingRule`](@ref) (defaults to `Uniform`) []
+ - `hypsography` the hypsography
+ - `global_geometry` the global geometry (defaults to `Geometry.CartesianGlobalGeometry`)
+ - `quad` the quadrature style (defaults to `Quadratures.GLL{n_quad_points}`)
+ - `h_mesh` the horizontal mesh (defaults to `Meshes.EquiangularCubedSphere`)
+ - `h_topology` the horizontal topology (defaults to `Topologies.Topology2D`)
+ - `horizontal_layout_type` the horizontal DataLayout type (defaults to `DataLayouts.IJFH`)
+ - `z_mesh` the z-mesh, defaults to an `Meshes.IntervalMesh` along `z` with given `stretch`
+ - `enable_bubble` enables the "bubble correction" for more accurate element areas.
+
```

Comment:
Can you add a `See also` section where you add `ref`s and links to the ClimaCore concepts used in this function, such as `Mesh`, `Domain`, `DataLayout`, `GlobalGeometry`, et cetera.

I think it would be useful to also add a few words to describe what they encode in the list of parameters, e.g.
```
 - `horizontal_layout_type` the horizontal DataLayout type (defaults to `DataLayouts.IJFH`). This parameter describes how data is arranged in memory and it relevant for performance.
```